### PR TITLE
WriteUnPrepared: Add new WAL marker kTypeBeginUnprepareXID

### PR DESCRIPTION
**Pull request by lth**
This adds a new WAL marker of type kTypeBeginUnprepareXID.

Also, DBImpl now contains a field called batch_per_txn (meaning one WriteBatch per transaction, or possibly multiple WriteBatches). This would also indicate that this DB is using WriteUnprepared policy.

Recovery code would be able to make use of this extra field on DBImpl in a separate diff. For now, it is just used to determine whether the WAL is compatible or not.

### CODE REVIEW

File: db/write_batch.cc [hunk 1]
```diff
@@ -1052,7 +1089,8 @@ class MemTableInserter : public WriteBatch::Handler {
                    bool ignore_missing_column_families,
                    uint64_t recovering_log_number, DB* db,
                    bool concurrent_memtable_writes,
-                   bool* has_valid_writes = nullptr, bool seq_per_batch = false)
+                   bool* has_valid_writes = nullptr, bool seq_per_batch = false,
+                   bool batch_per_trx = true)
```
Comment (maysamyabandeh) [inline db/write_batch.cc]: To be consistent lets use txn as the achronym for transaction.

File: db/write_callback_test.cc [hunk 2]
```diff
@@ -306,7 +307,7 @@ TEST_F(WriteCallbackTest, WriteWithCallbackTest) {
                   const size_t ONE_BATCH = 1;
                   s = db_impl->WriteImpl(
                       woptions, &write_op.write_batch_, &write_op.callback_,
-                      nullptr, 0, false, nullptr, ONE_BATCH,
+                      nullptr, 1, false, nullptr, ONE_BATCH,
```
Comment (maysamyabandeh) [inline db/write_callback_test.cc]: What is the reason for this change?

Comment (lth) [reply db/write_callback_test.cc]: This change was accidental.
